docusaurus/docs/shared/implement-resource-handler.md

```diff
@@ -0,0 +1,112 @@
+## Implement the resource handler interface
+
+To add a resource handler to your backend plugin, you need to implement the `backend.CallResourceHandler` interface.
+
+There are two ways you can implement this in your plugin, [using the `httpadapter` package](#using-the-httpadapter-package) or [manually implementing it](#manually-implementing-backendcallresourcehandler) in your plugin.
+
+### Using the `httpadapter` package
+
+The [`httpadapter`](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend/resource/httpadapter) package provided by the [Grafana Plugin SDK for Go](../../key-concepts/backend-plugins/grafana-plugin-sdk-for-go) is the recommended way for handling resources. This package provides support for handling resource calls using using the [`http.Handler`](https://pkg.go.dev/net/http#Handler) interface and allows responding to HTTP requests in a more Go-agnostic way and makes it easier to support multiple routes and methods (GET, POST etc).
+
+Using [`http.Handler`](https://pkg.go.dev/net/http#Handler) allows you to also use Go’s built-in router functionality called [`ServeMux`](https://pkg.go.dev/net/http#ServeMux) or your preferred HTTP router library (for example, [`gorilla/mux`](https://github.com/gorilla/mux)).
+
+:::note
+
+Go 1.22 [includes routing enhancement](https://go.dev/blog/routing-enhancements) that adds support for method matching and wildcards using the [`ServeMux`](https://pkg.go.dev/net/http#ServeMux).
+
+:::
+
+In the following example we demonstrate using the `httpadapter` package, `ServeMux` and `http.Handler` to add support for retrieving namespaces (`/namespaces`), projects (`/projects`) and updating the state of some device (`/device`) :
+
+```go
+package myplugin
+
+import (
+  "context"
+  "net/http"
+
+  "github.com/grafana/grafana-plugin-sdk-go/backend"
+  "github.com/grafana/grafana-plugin-sdk-go/backend/resource/httpadapter"
+)
+
+type MyPlugin struct {
+  resourceHandler backend.CallResourceHandler
+}
+
+func New() *MyPlugin {
+  p := &MyPlugin{}
+  mux := http.NewServeMux()
+  mux.HandleFunc("/namespaces", p.handleNamespaces)
+  mux.HandleFunc("/projects", p.handleProjects)
+  p.resourceHandler := httpadapter.New(mux)
+  return p
+}
+
+func (p *MyPlugin) CallResource(ctx context.Context, req *backend.CallResourceRequest, sender backend.CallResourceResponseSender) error {
+  return p.resourceHandler.CallResource(ctx, req)
```

Comment:
FTR, I just noticed that this had a typo, it should be:

```
  return p.resourceHandler.CallResource(ctx, req, sender)
```

Reply:
Thanks. Let me fix